### PR TITLE
Fixed issue when parsing ISO style dates (but not valid ISO format) with a trailing Z (Issue #510)

### PR DESCRIPTION
**Pull request by dorakung**
This PR fixed the issue [https://github.com/date-fns/date-fns/issues/510](url)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -96,6 +96,7 @@ for the list of changes made since `v2.0.0-alpha.1`.
 - [Support for long and relative formats for Swedish locale](https://github.com/date-fns/date-fns/pull/570) (thanks to [@alexandernanberg](https://github.com/alexandernanberg)).
 
 ### Changed
+- **BREAKING**: fixed when parsing ISO style dates (but not valid ISO format) with a trailing Z (e.g 2012-01Z), it returns Invalid Date for FireFox/IE11 (Issue #510)
```
Comment (leshakoss) [inline CHANGELOG.md]: It's not breaking, it's a bug fix

Comment (leshakoss) [reply CHANGELOG.md]: Also please convert issue number to a link (i.e. `[#510](https://github.com/date-fns/date-fns/issue/510)`)

File: src/toDate/test.js [hunk 2]
```diff
@@ -219,6 +219,12 @@ describe('toDate', function () {
           assert.deepEqual(result, new Date('2014-10-25T13:46:20+07:00'))
         })
       })
+      context('when the year and the month are specified', function () {
+        it('sets timezone correctly on yyyy-MMZ format', function () {
+          var result = toDate('2012-01Z')
+          assert.deepEqual(result, new Date('2011-12-31T18:00:00+07:00'))
```
Comment (leshakoss) [inline src/toDate/test.js]: @dorakung please make it `'2012-01-01T00:00:00+00:00'`. It is the same thing, but shows the intention clearer

**Comment by richie50** [comment]
image



Results =>
image

parse routine does not work on the latest firefox. Thank @leshakoss

**Comment by leshakoss** [comment]
@richie50 I am not sure what you're trying to do. Could you please elaborate? (I don't see any connection between your two screenshots) And also could you please create a new issue for that?

**Comment by richie50** [comment]
@leshakoss for some reason on firefox the dateString Object with the field _date: "07-May-2018"_ can not be converted properly to a date object by the routine **_parseDate(arg1, arg2)._** . **(Refer to the above code)**.I did trace the code and i noticed on firefox the new Date("07-May-2018") _returns null_ but this does not happen on chrome. For now i'm going to change the  "07-May-2018" to an acceptable format that can be parsed by the Javascript Date Object. Thanks.

**Comment by leshakoss** [comment]
@richie50 if the string passed to `parse` fn is not in the [ISO-8601](https://en.wikipedia.org/wiki/ISO_8601) format, then it falls back to `new Date` constructor, which could work differently in different browsers

**Comment by richie50** [comment]
@leshakoss thanks. 👍

**Pull request merged**